code/parse/sexp_container.cpp

```diff
@@ -329,6 +337,233 @@ sexp_container *get_sexp_container_special(const SCP_string& text, size_t start_
 	return nullptr;
 }
 
+/**
+* Helper function for sexp_container_replace_refs_with_values(). Given a SEXP Container it works out what modifer was used and what the replacement string should be.
+**/
+bool get_replace_text_for_modifier(const SCP_string &text,
+	sexp_container &container,
+	const size_t lookHere,
+	SCP_string &replacement_text,
+	size_t &num_chars_to_replace)
+{
+	// for map containers, check if this matches a map key
+	if (container.is_map()) {
+		if (container.map_data.empty()) {
+			Warning(LOCATION, "Attempt to replace text using empty map container %s", container.container_name.c_str());
+			return false;
+		}
+
+		const size_t key_ends_here = text.find(sexp_container::DELIM, lookHere);
+		if (key_ends_here == SCP_string::npos) {
+			Warning(LOCATION,
+				"sexp_container_replace_refs_with_values() found a map container called %s to replace but the key is "
+				"missing a final '&'",
+				container.container_name.c_str());
+			return false;
+		}
+
+		const SCP_string key = text.substr(lookHere, key_ends_here - lookHere);
+		const auto iter = container.map_data.find(key);
+
+		if (iter == container.map_data.end()) {
+			Warning(LOCATION,
+				"sexp_container_replace_refs_with_values() found a map container called %s but the key %s is not found",
+				container.container_name.c_str(),
+				key.c_str());
+			return false;
+		}
+
+		replacement_text = iter->second;
+
+		num_chars_to_replace += key.length();
+	} else {
+		// for list containers, check if this matches a list modifier
+		Assert(container.is_list());
+
+		if (container.list_data.empty()) {
+			Warning(LOCATION, "Attempt to replace text using empty list container %s", container.container_name.c_str());
+			return false;
+		}
+
+		const auto modifier = get_list_modifier(text.c_str() + lookHere, true);
+
+		if (modifier == ListModifier::INVALID) {
+			Warning(LOCATION,
+				"sexp_container_replace_refs_with_values() found a list container called %s to replace but the modifer "
+				"is not recognised",
```

Comment:
Again, I think I would be mystified if I saw this as a modder, and I wanted to figure out what to do.